Bearded.Utilities.Testing.Tests/Core/MaybeAssertionsTests.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using FluentAssertions;
+using Xunit;
+using Xunit.Sdk;
+
+namespace Bearded.Utilities.Testing.Tests
+{
+    public class MaybeAssertionsTests
+    {
+        [Fact]
+        public void BeJust_SucceedsWhenMaybeHasValue()
+        {
+            var maybe = Maybe.Just(100);
+
+            Action assertion = () => maybe.Should().BeJust();
+            
+            assertion.Should().NotThrow();
+        }
+        
+        [Fact]
+        public void BeJust_FailsWhenMaybeIsNothing()
+        {
+            var maybe = Maybe.Nothing<int>();
+
+            Action assertion = () => maybe.Should().BeJust();
+            
+            assertion.Should().Throw<XunitException>();
+        }
+        
+        [Fact]
+        public void BeJust_SucceedsWithAndConstraint()
+        {
+            var maybe = Maybe.Just(100);
+
+            Action assertion = () => maybe.Should().BeJust().Which.Should().Be(100);
+            
+            assertion.Should().NotThrow();
+        }
+        
+        [Fact]
+        public void BeJust_FailsWhenAndConstraintIsNotMet()
+        {
+            var maybe = Maybe.Just(100);
+
+            Action assertion = () => maybe.Should().BeJust().Which.Should().Be(200);
+            
+            assertion.Should().Throw<XunitException>();
+        }
+
+        [Fact]
+        public void BeJust_SucceedsWhenComparingToMaybeWithRightValue()
```

Comment:
It's comparing to the right value, not to a maybe with the right value, no?

Reply:
It took me a while to understand this comment, but yeah, you're right. Updated all the test names to hopefully be clearer.